docs/docs/explainers/explainer-writing-noir.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Writing Noir
+description: Understand new considerations when writing Noir
+keywords: [Noir, programming, rust]
+tags: [Optimization]
+sidebar_position: 0
+---
+
+# Writing Noir for fun and profit
+
+This article intends to set you up with a key concept essential for writing more viable applications that use zero knowledge proofs, namely around non-wasteful circuits.
+
+## Context - 'Efficient' is subjective
+
+When writing a web application for a performant computer with high-speed internet connection, writing efficient code sometimes is seen as an afterthought only if needed. Large multiplications running at the innermost of nested loops may not even be on a dev's radar.
+When writing firmware for a battery-powered microcontroller, you think of cpu cycles as rations to keep within a product's power budget.
+
+> Code is written to create applications that perform specific tasks within specific constraints
+
+And these constraints differ depending on where the compiled code is execute.
+### The Ethereum Virtual Machine (EVM)
```

Comment:
Do you think this context about the EVM is value-additive? Just thinking "less is more" and it could be more distracting than helpful.

Reply:
Agreed, will reduce/refine this.